include/alpaka/rand/RandPhilox.hpp

```diff
@@ -0,0 +1,232 @@
+#include <alpaka/rand/Philox/PhiloxVector.hpp>
+#include <alpaka/rand/Traits.hpp>
+
+#include <cstdint>
+#include <limits>
+#include <random>
+#include <type_traits>
+
+namespace alpaka
+{
+    namespace rand
+    {
+        /** Most common Philox engine variant, outputs single number
+         *
+         * This is a variant of the Philox engine generator which outputs a single float. The counter size is \f$4
+         * \times 32 = 128\f$ bits. Since the engine returns a single number, the generated result, which has the same
+         * size as the counter, has to be stored between invocations. Additionally a 32 bit pointer is stored. The
+         * total size of the state is 352 bits = 44 bytes.
+         *
+         * Ref.: J. K. Salmon, M. A. Moraes, R. O. Dror and D. E. Shaw, "Parallel random numbers: As easy as 1, 2, 3,"
+         * SC '11: Proceedings of 2011 International Conference for High Performance Computing, Networking, Storage and
+         * Analysis, 2011, pp. 1-12, doi: 10.1145/2063384.2063405.
+         *
+         * @tparam TAcc Accelerator type as defined in alpaka/acc
+         */
+        template<typename TAcc>
+        class Philox4x32x10 : public concepts::Implements<ConceptRand, Philox4x32x10<TAcc>>
+        {
+        public:
+            using EngineParams
+                = engine::PhiloxParams<4, 32, 10>; ///< Philox algorithm: 10 rounds, 4 numbers of size 32.
+            using EngineVariant = engine::PhiloxSingle<TAcc, EngineParams>; ///< Engine outputs a single number
+
+            /** Initialize a new Philox engine
+             *
+             * @param seed Set the Philox generator key
+             * @param subsequence Select a subsequence of size 2^64
+             * @param offset Skip \a offset numbers form the start of the subsequence
+             */
+            ALPAKA_FN_HOST_ACC Philox4x32x10(
+                std::uint64_t const seed = 0,
+                std::uint64_t const subsequence = 0,
+                std::uint64_t const offset = 0)
+                : engineVariant(seed, subsequence, offset)
+            {
+            }
+
+            ALPAKA_FN_HOST_ACC Philox4x32x10(EngineVariant state) : engineVariant(state)
+            {
+            }
+
+            EngineVariant engineVariant;
+
+            // STL UniformRandomBitGenerator concept
+            // https://en.cppreference.com/w/cpp/named_req/UniformRandomBitGenerator
+            using result_type = std::uint32_t;
+            ALPAKA_FN_HOST_ACC constexpr result_type min()
+            {
+                return 0;
+            }
+            ALPAKA_FN_HOST_ACC constexpr result_type max()
+            {
+                return std::numeric_limits<result_type>::max();
+            }
+            ALPAKA_FN_HOST_ACC result_type operator()()
+            {
+                return engineVariant();
+            }
+            // TODO: https://en.cppreference.com/w/cpp/named_req/RandomNumberEngine
```

Comment:
Do you mean that the remaining features of the `RandomNumberEngine` concepts are missing? If yes, please say so here.